content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,401 @@ preview: true
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `Cascade`?
+
+If a `User` record is deleted, then their posts are deleted too. If the user's `id` is updated, then the corresponding `authorId` is also updated.
+
+### Restrict
+
+- `onDelete: Restrict` Prevents the deletion if any referencing records exist.
+
+- `onUpdate: Restrict` Prevents the identifier of a referenced record from being changed.
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Restrict, onUpdate: Restrict)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `Restrict`?
+
+`User`'s with posts **cannot** be deleted. The `Post`'s `id` **cannot** be changed.
+
+<Admonition type="warning">
+
+The `Restrict` action is **not** available on [SQL Server](../../../database-connectors/microsoft-sql-server) and will trigger a schema validation error. Instead, you can use [`NoAction`](#noaction), which produces the same result and is compatible with SQLServer.
+
+</Admonition>
+
+### NoAction
+
+- `onDelete: NoAction` Prevents the deletion if any referencing records exist.
+
+- `onUpdate: NoAction` Prevents the identifier of a referenced record from being changed.
+
+The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependant on the database being used.
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `NoAction`?
+
+`User`'s with posts **cannot** be deleted. The `Post`'s `id` **cannot** be changed.
+
+### SetNull
+
+- `onDelete: SetNull` The scalar field of the referenced object will be set to `NULL`.
+
+- `onUpdate: SetNull` When updating the identifier of a referenced object, the scalar fields of the referencing objects will be set to `NULL`.
+
+If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull, onUpdate: SetNull)
+  authorId  Int
```

Comment:
```suggestion
  author    User?    @relation(fields: [authorId], references: [id], onDelete: SetNull, onUpdate: SetNull)
  authorId  Int?
```

I would recommend making the relation optional in the example, otherwise it would throw a runtime error as it's been called out above.